src/formatDate.js

```diff
@@ -8,7 +8,53 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  let fromSeparator = '';
+
+  for (const arg of fromFormat) {
+    if (arg !== 'YYYY' || arg !== 'YY' || arg !== 'MM' || arg !== 'DD') {
+      fromSeparator = arg;
+    }
+  }
+
+  const dateParts = date.split(fromSeparator);
+
+  let year = '';
+  let month = '';
+  let day = '';
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    if (fromFormat[i] === 'YYYY' || fromFormat[i] === 'YY') {
+      year = dateParts[i];
+    } else if (fromFormat[i] === 'MM') {
+      month = dateParts[i];
+    } else {
+      day = dateParts[i];
+    }
+  }
+
+  if (toFormat.includes('YYYY') && fromFormat.includes('YY')) {
+    year = year < 30 ? `20${year}` : `19${year}`;
```

Comment:
The logic for converting 'YY' to 'YYYY' assumes that years less than 30 belong to the 2000s. This assumption might not be valid for all scenarios. Consider making this logic configurable or more robust to handle different cases.

Reply:
This was also stated in the problem statement.
The function can change a separator, reorder the date parts of convert a year from 4 digits to 2 digits and back.

When converting from YYYY to YY just use 2 last digit (1997 -> 97).
When converting from YY to YYYY use 20YY if YY < 30 and 19YY otherwise.